packages/aws-client/src/S3ObjectStore.ts

```diff
@@ -0,0 +1,100 @@
+import aws from 'aws-sdk';
+import * as querystring from 'querystring';
+import { URL } from 'url';
+import { inTestMode } from '@cumulus/common/test-utils';
+import { s3 } from './services';
+import { headObject, parseS3Uri } from './S3';
+
+/**
+ * Class to use when interacting with S3
+ *
+ */
+export class S3ObjectStore {
+  s3: aws.S3;
+
+  /**
+   * Builds and returns an S3 object store configured according to environment variables
+   * Will use localstack if USE_LOCALSTACK is true (default false) and AWS_DEFAULT_REGION
+   * (default "us-west-2")
+   *
+   * @param {Object} overrides - values to set when constructing the underlying S3 store
+   */
+  constructor(overrides?: object) {
+    this.s3 = this._getS3(overrides);
```

Comment:
Sorry, I should have been more clear in my feedback. I meant something a bit simpler like this:

```suggestion
  constructor(overrides?: object) {
    this.s3 = overrides.s3 || s3();
```

And then remove the `_getS3` method entirely. I'm wondering if adding the `@cumulus/common` dependency to this package is part of the problem (though I can't see why). Still, I don't love having to repeat that logic for setting up the S3 service that we already have in this aws-client package

Reply:
Just to summarize from Slack:
Removing the `@cumulus/common` package fixed the build errors
Since `s3()` handles test conditions (and for ease) I removed the `_getS3` method and the `overrides` param. While we're still staging things to s3 this way, we're no longer using sinon unnecessarily.